src/psyclone/gocean1p0.py

```diff
@@ -1304,3 +1361,30 @@ def nargs(self):
     def index_offset(self):
         ''' Return the grid index-offset that this kernel expects '''
         return self._index_offset
+
+
+class GOACCDataDirective(ACCDataDirective):
+    '''
+    Sub-classes ACCDataDirective to provide an API-specific implementation
+    of data_on_device().
+
+    '''
+    def data_on_device(self, parent):
+        '''
+        Adds nodes into a Schedule to flag that each of the objects required
```

Comment:
We're not adding nodes to the schedule here. We're adding fortran as part of the fortran generation step.